src/csharp/BUILD-INTEGRATION.md

```diff
@@ -485,6 +489,42 @@ then in your project file:
 ```
 Do not include any `<Protobuf>` items in the project file as that will invoke the `Grpc.Tools` build and your files will be compiled twice.
 
+---
+
+## <a name="unsupported-arch"></a>Using Grpc.Tools with unsupported architectures
+
+You can still use the MSBuild integration provided by `Grpc.Tools` for architectures where
+the binaries are not included in the `Grpc.Tools` NuGet package.
+
+If you are able to build your own binaries for `protoc` and `grpc_csharp_plugin`, or
+find pre-built binaries provided by the commuity, then you can define a couple of
+environment variables to tell `Grpc.Tools` to use those binaries.
+
+Note: You need `Grpc.Tools` version 2.50.0 or later for these environment variables to
+be recognised.
+
+For example, there are community provided packages for the gRPC plugins for Alpine Linux using the *musl* C standard library:
+[https://pkgs.alpinelinux.org/packages?name=grpc-plugins](https://pkgs.alpinelinux.org/packages?name=grpc-plugins)
+
+To use these:
+```bash
+# Build or install the binaries for your architecture.
+
+# e.g. for Alpine Linux the grpc-plugins package can be used
+#        See https://pkgs.alpinelinux.org/package/edge/community/x86_64/grpc-plugins
+apk add grpc-plugins  # Alpine Linux specific package installer
```

Comment:
add a note somewhere that when providing your own binaries, you're also responsible they're the right version of protobuf a grpc plugin.
The binaries bundled in Grpc.Tools already ensure that the correct and mutually compatible version of `protoc` and `grpc_csharp_plugin` will be chosen, but when providing them yourself, you're in charge.
- protoc and grpc_csharp_plugin can be mutually incompatible
- the code generated by the wrong version of many not work with your application (e.g. breaks the build).
- specifically, older version of plugins may generate incompatible code or may not contain patches/fixes.